lib/widgets/compose_box.dart

```diff
@@ -953,24 +953,18 @@ class _FixedDestinationComposeBoxState extends State<_FixedDestinationComposeBox
 }
 
 class ComposeBox extends StatelessWidget {
-  const ComposeBox({super.key, this.controllerKey, required this.narrow});
+  const ComposeBox({super.key, this.controllerKey, required Narrow narrow}) : _narrow = narrow;
 
+  final Narrow _narrow;
   final GlobalKey<ComposeBoxController>? controllerKey;
-  final Narrow narrow;
 
   @override
   Widget build(BuildContext context) {
-    final narrow = this.narrow;
-    if (narrow is StreamNarrow) {
```

Comment:
narrow is refactored to become private because otherwise `switch` is not
    able to promote the type of it.

No need to change the class's definition; this local variable was solving the same problem already.